### ISSUE TITLE
Couldn't compile rocksdb library with gtest

### ISSUE THREAD

**Issue by ouliuquan** [facebook/rocksdb#6561]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://groups.google.com/forum/#!forum/rocksdb or https://www.facebook.com/groups/rocksdb.dev

### Expected behavior
I can not make the library compile finish, it was stop when error occuor.
### Actual behavior
[ 61%] Building CXX object tools/CMakeFiles/sst_dump.dir/sst_dump.cc.o
/usr/bin/ld: third-party/gtest-1.8.1/fused-src/gtest/libgtest.a(gtest-all.cc.o): undefined reference to symbol 'pthread_key_delete@@GLIBC_2.2.5'
/lib/x86_64-linux-gnu/libpthread.so.0: 无法添加符号: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/table_reader_bench.dir/build.make:98: recipe for target 'table_reader_bench' failed
make[2]: *** [table_reader_bench] Error 1
CMakeFiles/Makefile2:6774: recipe for target 'CMakeFiles/table_reader_bench.dir/all' failed
make[1]: *** [CMakeFiles/table_reader_bench.dir/all] Error 2
make[1]: *** 正在等待未完成的任务....
[ 61%] Linking CXX executable sst_dump
[ 61%] Built target sst_dump
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

### Steps to reproduce the behavior
mkdir build&& cd build
cmake  ..
make

**Comment by syoliver** [comment]
This issue should have been closed on master by #6572

**Comment by wouterbeek** [comment]
I have the same issue, but with gflags i.o. gtest:

```
$ cmake --build .
[518/556] Linking CXX executable cache_bench
FAILED: cache_bench 
: && ccache /usr/bin/c++  -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -Wno-unused-variable -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers -Wno-strict-aliasing -fno-omit-frame-pointer -momit-leaf-frame-pointer -march=native -Werror -fno-builtin-memcmp -O3 -DNDEBUG -fno-rtti   CMakeFiles/cache_bench.dir/cache/cache_bench.cc.o  -o cache_bench  -Wl,-rpath,/home/wouter/github/rocksdb/Release  librocksdb.so.6.10.0 && :
/usr/bin/ld: CMakeFiles/cache_bench.dir/cache/cache_bench.cc.o: undefined reference to symbol '_ZN6google14FlagRegistererC1IbEEPKcS3_S3_PT_S5_'
/usr/bin/ld: /lib/x86_64-linux-gnu/libgflags.so.2.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
[531/556] Building CXX object CMakeFiles/db_bench.dir/tools/db_bench_tool.cc.o
ninja: build stopped: subcommand failed.

**Comment by ajkr** [comment]
@wouterbeek Should be fixed by #7011. Let us know if you still see the issue.
